appmap/_implementation/recorder.py

```diff
@@ -147,13 +163,23 @@ def _stop_recording(self):
     def _add_event(self, event):
         super()._add_event(event)
 
+    # pragma pylint: enable=useless-super-delegation
+
 
 class SharedRecorder(Recorder):
     """
     A shared Recorder. The global recorder is an instance of this class.
     """
 
-    _lock = threading.Lock()
+    _lock = threading.RLock()
```

Comment:
It may be worth explaining in a comment how the non re-entrancy lead to what this suggests to have been a deadlock.  How did a thread, after acquiring this lock, called code that tried to acquire the lock again?

Reply:
Good catch. Using an `RLock` here was a holdover from a previous attempt to fix this problem. It's no longer necessary, so I've switched back to `Lock`.